src/cascadia/Remoting/WindowManager.cpp

```diff
@@ -118,78 +119,78 @@ namespace winrt::Microsoft::Terminal::Remoting::implementation
 
                 proposedCommandline = true;
             }
-            catch (const winrt::hresult_error& e)
+            catch (...)
             {
-                // We did not successfully ask the king what to do. They
-                // hopefully just died here. That's okay, let's just go ask the
-                // next in the line of succession. At the very worst, we'll find
-                // _us_, (likely last in the line).
-                //
-                // If the king returned some _other_ error here, than lets
-                // bubble that up because that's a real issue.
+                // We did not successfully ask the king what to do. This could
+                // be for many reasons. Most commonly, the monarch died as we
+                // were talking to it. That could be a RPC_SERVER_UNAVAILABLE_HR
+                // or RPC_CALL_FAILED_HR (GH#12666). We also saw a
+                // RPC_S_CALL_FAILED_DNE in GH#11790. Ultimately, if this is
+                // gonna fail, we want to just try again, regardless of the
+                // cause. That's why we're no longer checking what the exception
+                // was, we're just always gonna try again regardless.
                 //
-                // I'm checking both these here. I had previously got a
-                // RPC_S_CALL_FAILED about here once.
-                if (e.code() == RPC_SERVER_UNAVAILABLE_HR || e.code() == RPC_CALL_FAILED_HR)
+                //  They hopefully just died here. That's okay, let's just go
+                // ask the next in the line of succession. At the very worst,
+                // we'll find _us_, (likely last in the line).
+                TraceLoggingWrite(g_hRemotingProvider,
+                                  "WindowManager_proposeToMonarch_unexpectedExceptionFromKing",
+                                  TraceLoggingInt32(attempts, "attempts", "How many times we've tried"),
+                                  TraceLoggingLevel(WINEVENT_LEVEL_VERBOSE),
+                                  TraceLoggingKeyword(TIL_KEYWORD_TRACE));
+                LOG_CAUGHT_EXCEPTION();
+                attempts++;
+
+                if (attempts >= 10)
                 {
+                    // We've tried 10 times to find the monarch, failing each
+                    // time. Since we have no idea why, we're guessing that in
+                    // this case, there's just a Monarch registered that's
+                    // misbehaving. In this case, just fall back to
+                    // "solatedMonarchMode" - we can't trust the currently

```

Comment:
```suggestion
                    // "isolatedMonarchMode" - we can't trust the currently
```
Is that what you meant? Also curious, what happens in "isolatedMonarchMode"? Do we assume our monarch has died and elect a new one? Do we need to worry about that dead monarch suddenly being alive again when there's a newly elected monarch? (this could be a great TV show haha)

Reply:
More or less - whoever the monarch is, they are clearly misbehaving, so we choose to ignore them. We declare ourselves king, and all calls from us to the monarch go to us instead.

Now, if there's a well-intentioned monarch that other windows are communicating with, then this new window will never be in contact with them. 

If there truly is a bad monarch, then future windows will also hit this case, and be permanently isolated from one another. Alas, this is preferable to the windows _dying_.

I'm sure there's a real world historical metaphor for this, but I worry I'll step on some geopolitical toes somewhere so let's just call this the _War of the Five Kings Mode_.